cmd/server/main.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/go-kit/kit/log"
+	"github.com/redhatinsights/insights-operator-conditional-gathering/pkg/service"
+	"github.com/redhatinsights/insights-operator-conditional-gathering/pkg/service/transport"
+)
+
+const (
+	httpAddr = ":8080"
+)
+
+func main() {
+	// Logger
+	var logger log.Logger
+	logger = log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	logger = log.With(logger, "ts", log.DefaultTimestampUTC)
+
+	// Repository
+	dataPath := os.Getenv("RULES_DATA_PATH")
+	repo := service.NewRepository(dataPath)
+
+	// Services
+	svc := service.New(repo)
+
+	httplogger := log.With(logger, "component", "http")
+
+	mux := http.NewServeMux()
+	// Control CORs and other stuff
+	http.Handle("/", accessControl(mux))
+
+	// Handlers
+	mux.Handle("/", transport.NewHTTPHandler(svc, httplogger))
+
+	// HTTP
+	errs := make(chan error, 2)
+
+	go func() {
+		logger.Log("transport", "http", "address", httpAddr, "msg", "listening") // nolint: errcheck
+		errs <- http.ListenAndServe(httpAddr, nil)
+	}()
+
+	go func() {
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, syscall.SIGINT, syscall.SIGTERM)
+		errs <- fmt.Errorf("%s", <-c)
+	}()
+
+	logger.Log("terminated", <-errs) // nolint: errcheck
```

Comment:
Shouldn't we close the `errs` channel? It seems weird to have the channel with bufer size 2 and read only once from the channel.

Reply:
OK I guess I've learnt the capacity makes sense here, because otherwise it could probably end up in a blocked state (sending to a channel with full buffer), but I would probably prefer to use two channels anyway.......just my 2 cents.

Reply:
I think that the last commit should solve it.